core/src/test/java/org/infinispan/test/TestingUtil.java

```
@@ -338,6 +340,14 @@ public static Object extractField(Class type, Object target, String fieldName) {
             .findInterceptorExtending(interceptorToFind);
    }
 
+   public static int getSegmentForKey(Object key, Cache cache) {
+      KeyPartitioner keyPartitioner = extractComponent(cache, KeyPartitioner.class);
+      if (keyPartitioner != null) {
+         return keyPartitioner.getSegment(key);
+      }
+      return SegmentSpecificCommand.UNKNOWN_SEGMENT;
```

I'd rather throw an exception

---

Okay